addons/misra.py

```diff
@@ -256,14 +256,105 @@ def getForLoopExpressions(forToken):
             lpar.astOperand2.astOperand2.astOperand2]
 
 
+def findCounterTokens(cond):
+    if cond.str in ['&&', '||']:
+        c = findCounterTokens(cond.astOperand1)
+        c.extend(findCounterTokens(cond.astOperand2))
+        return c
+    ret = []
+    if cond.isArithmeticalOp:
+        if cond.astOperand1.isName and cond.astOperand2.isNumber:
```

Comment:
It seems you do not handle unary operators. As far as I know '+' will mean that `isArithmeticalOp` is true no matter if it is unary or binary.